sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedParquetRecordReader.java

```diff
@@ -197,11 +200,17 @@ public void initBatch(MemoryMode memMode, StructType partitionColumns,
   }
 
   public void initBatch() {
-    initBatch(DEFAULT_MEMORY_MODE, null, null);
+    assert(memoryMode != null);
+    initBatch(memoryMode,null, null);
```

Comment:
If `initBatch()` is called after calling `initBatch(..., ..., true), we have to use `MemoryMode.ON_HEAP` instead of `SparkEnv.get().memoryManager().tungstenMemoryMode()`.